test/models/profile_test.rb

```diff
@@ -863,4 +863,40 @@ class ProfileTest < ActiveSupport::TestCase
       assert_equal [@rule1], @profile.removed_rules
     end
   end
+
+  context 'rules_and_rule_groups' do
+    setup do
+      @parent = FactoryBot.create(
+        :canonical_profile,
+        :with_rules,
+        rule_count: 4
+      )
+
+      @rule1 = @parent.rules.first
+      @rule2 = @parent.rules.second
+      @rule3 = @parent.rules.third
+      @rule4 = @parent.rules.fourth
+
+      @rule_group_1 = FactoryBot.create(:rule_group)
+      @rule_group_2 = FactoryBot.create(:rule_group)
+      @rule_group_3 = FactoryBot.create(:rule_group)
+
```

Comment:
@marleystipich2 I'm missing records in `rule_group_rules` connecting the records from `rule_groups` and `rules`.